### ISSUE TITLE
Switch Linker Libraries

### ISSUE THREAD

**Issue by ThePhaseCat** [ThePhaseCat/phases-discord-rich-presence#9]
The current linker/binding library is outdated and hasn't been updated since 2021. As such, it's missing some stuff and support for arm-based devices. The best course of action would be to migrate, and it seems like this repo (https://github.com/JnCrMx/discord-game-sdk4j) is the best bet for now as it's actually up-to-date and maintained. 

This would be for the next update, and this would take a while as I basically need to recode/change all of the discord related stuff. Due to how big of an update this would be, I'll be working on this in a separate branch which will also be separate from any 1.6 related work I want to do in the near future.

**Comment by DeveloLongScript** [comment]
Maybe try using something like [KDiscordIPC](https://github.com/caoimhebyrne/KDiscordIPC), used in the Essential mod as its both multi-platform and supports every feature, including grabbing somebodies username and discriminator when the IPC starts

**Comment by ThePhaseCat** [comment]
I've been pretty busy this week because of midterms but I was messing around on Monday and I was unsuccessful in getting either library working properly.

**Comment by ThePhaseCat** [comment]
update, it seems like https://github.com/JnCrMx/discord-game-sdk4j this library might actually work and i was coding it wrong? this would require a massive rewrite of the project, but would be quite worth it for the guaranteed support it would bring for all platforms

**Comment by DeveloLongScript** [comment]
Maybe make a class like `MigrationDiscord` which has all of the old function names with the new library, so you wouldn't have to re-write everything.

**Comment by ThePhaseCat** [comment]
yeah i was thinking of rewriting the code so this would give me an excuse to do it lol

probably something for after college finals

**Comment by ThePhaseCat** [comment]
well, got a super basic version working already using a new library, will move forward into what is basically a rewrite of most of the mod lol

![image](https://github.com/user-attachments/assets/f7276dba-e173-433d-a665-e915f3f83a9f)

**Comment by ThePhaseCat** [comment]
(and yes, i figured out how to get player heads to work with the help of another repo i saw!)

**Comment by ThePhaseCat** [comment]
![image](https://github.com/user-attachments/assets/261c0c4c-aa15-4e76-984c-15f3f50e7a3b)

this is a problem

**Comment by ThePhaseCat** [comment]
![image](https://github.com/user-attachments/assets/e364abbb-9456-4a4c-90a4-15dfd170e9fe)
update

it seems to build and function as expected so it might just be that github actions is doing something weird

**Comment by ThePhaseCat** [comment]
beta version with this new library has been released
https://github.com/ThePhaseCat/phases-discord-rich-presence/releases/tag/v2.0.0-beta1
https://modrinth.com/mod/phases-discord-rich-presence/version/2.0.0-1.21-beta1

will close this issue once the entire update is done

**Comment by ThePhaseCat** [comment]
Done with v2.0.0, closing issue now!

**Comment by DeveloLongScript** [comment]
Awesome!

**Comment by DeveloLongScript** [comment]
Awesome!
